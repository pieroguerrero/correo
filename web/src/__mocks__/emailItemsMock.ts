import { IEmailListDTO } from "../dtos";

const uniqueEmailItem: IEmailListDTO = {
  _id: "1",
  isActive: true,
  isRead: false,
  created: new Date().toISOString(),
  received: new Date().toISOString(),
  sent: "",
  from: "nigerianprince@email.com",
  senderName: "Bakare Tunde",
  to: "myemail@email.com",
  cc: "ccemail1@email.com, ccemail2@email.com",
  bcc: "",
  subject: "An importan donation for you",
  subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
  body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
  priority: "1",
  type: "text",
};

const wrongEmailItem: IEmailListDTO = {
  _id: "2",
  isActive: true,
  isRead: false,
  created: new Date().toISOString(),
  received: new Date().toISOString(),
  sent: "",
  from: "nigerianprinceemail.com",
  senderName: "Bakare Tunde",
  to: "myemail@email.com",
  cc: "ccemail1@email.com, ccemail2@email.com",
  bcc: "",
  subject: "An importan donation for you",
  subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
  body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
  priority: "1",
  type: "html",
};

const emailAddress = "myemail@email.com";

const emailItemsList: IEmailListDTO[] = [
  {
    _id: "3",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
  {
    _id: "4",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
  {
    _id: "5",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
  {
    _id: "6",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
  {
    _id: "7",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
  {
    _id: "8",
    isActive: true,
    isRead: false,
    created: "2023-03-15T17:10:35.961Z",
    received: "2023-03-15T17:10:35.961Z",
    sent: "",
    from: "nigerianprince@email.com",
    senderName: "Bakare Tunde",
    to: "myemail@email.com",
    cc: "ccemail1@email.com, ccemail2@email.com",
    bcc: "",
    subject: "An importan donation for you",
    subjectDetails: "REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL",
    body: "I am Dr. Bakare Tunde, the cousin of Nigerian Astronaut, Air Force Major Abacha Tunde...",
    priority: "1",
    type: "text",
  },
];

export { uniqueEmailItem, wrongEmailItem, emailItemsList, emailAddress };
