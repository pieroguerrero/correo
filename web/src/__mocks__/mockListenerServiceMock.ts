import { IEmailListDTO } from "../dtos";

let callbackEvent: null | ((newEmail: IEmailListDTO) => void) = null;
const mockListenerService = {
  on: (_eventName: string, callback: (newEmail: IEmailListDTO) => void) => {
    callbackEvent = callback;
  },
  emit: (_eventName: string, email: IEmailListDTO) => {
    if (callbackEvent) {
      callbackEvent(email);
    }
  },
  off: (_eventName: string) => {
    callbackEvent = null;
    return;
  },
};

export default mockListenerService;
