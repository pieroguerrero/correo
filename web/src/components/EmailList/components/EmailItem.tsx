import { IEmail } from "../../../interfaces";

interface IEmailItem {
  email: IEmail;
}
export default function EmailItem({ email }: IEmailItem) {
  return (
    <li className="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100">
      <div>
        <input type="checkbox" />
      </div>
      <button className="flex flex-row items-center space-x-4 max-w-full">
        <div>
          <button>
            <svg
              className="w-4 h-4 text-gray-500"
              viewBox="0 0 24 24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                stroke="currentColor"
                strokeWidth="2"
                strokeLinecap="round"
                strokeLinejoin="round"
                fill="none"
              ></path>
            </svg>
          </button>
        </div>
        <div>
          <div className="w-36 max-w-[9rem]">
            <p className="text-sm">
              {email.fromFullName ? email.fromFullName : "No Name"}
            </p>
          </div>
        </div>
        <div className="flex-1">
          <p className="text-sm">{email.subject}</p>
        </div>
        <div>
          {/* (new Date(item.data.created * 1000)).toLocaleString() */}
          <p className="text-sm">{email.created.toLocaleString()}</p>
        </div>
      </button>
    </li>
  );
}
