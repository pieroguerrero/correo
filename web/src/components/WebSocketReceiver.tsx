import { useContext, useEffect, useState } from "react";
import { WebSocketContext } from "../contexts/WebSocketContext";

//Check this video to know how to configure the websocket server the cors to allow this client to get the information from websocket:https://youtu.be/7mm2xUzjomk?t=771
export function WebSocketReceiver() {
  const [newEmail, setNewEmail] = useState(null);
  const socket = useContext(WebSocketContext);
  useEffect(() => {
    socket.on("newEmail", (email) => {
      setNewEmail(email);
      console.log(email);
    });
    return () => {
      socket.off("newEmail");
    };
  }, []);
  return (
    <div className=" text-red-600">
      {newEmail ? newEmail["from"] : "No messages"}
    </div>
  );
}
