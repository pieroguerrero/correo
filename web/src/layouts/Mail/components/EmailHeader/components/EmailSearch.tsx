export default function EmailSearch() {
  return (
    <div className=" h-12 px-4 border-0 rounded-xl bg-blue-100/50 my-2 flex items-center space-x-4">
      <div>
        <svg
          focusable="false"
          viewBox="0 0 24 24"
          className="w-6 h-6 text-gray-700"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill="currentColor"
            d="M20.49,19l-5.73-5.73C15.53,12.2,16,10.91,16,9.5C16,5.91,13.09,3,9.5,3S3,5.91,3,9.5C3,13.09,5.91,16,9.5,16 c1.41,0,2.7-0.47,3.77-1.24L19,20.49L20.49,19z M5,9.5C5,7.01,7.01,5,9.5,5S14,7.01,14,9.5S11.99,14,9.5,14S5,11.99,5,9.5z"
          ></path>
          <path d="M0,0h24v24H0V0z" fill="none"></path>
        </svg>
      </div>
      <input
        placeholder="Search in mail"
        className="flex-1 bg-transparent outline-none placeholder-gray-600"
      />
      <div>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="w-6 h-6 text-gray-700"
          viewBox="0 0 24 24"
        >
          <path
            fill="currentColor"
            d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
          ></path>
        </svg>
      </div>
    </div>
  );
}
