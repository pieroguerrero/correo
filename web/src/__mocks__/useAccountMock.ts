import { IUserAccount } from "../models/UserAccount";
import { emailAddress } from "./emailItemsMock";

const useAccountMock: IUserAccount = {
  name: "No Name",
  lastname: "No Last Name",
  email: emailAddress,
};

export { useAccountMock };
