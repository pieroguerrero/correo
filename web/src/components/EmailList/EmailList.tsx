import { IEmail } from "../../interfaces";
import EmailItem from "./components/EmailItem";
interface IEmailList {
  emails: IEmail[];
}

/**
 * Component that render the emails.
 * @returns
 */
export function EmailList({ emails }: IEmailList) {
  //const userAccount = useContext(UserAccountContext);
  return (
    <div>
      <ul>
        {emails.length === 0
          ? null
          : emails.map((email) => <EmailItem key={email._id} email={email} />)}
      </ul>
    </div>
  );
}
