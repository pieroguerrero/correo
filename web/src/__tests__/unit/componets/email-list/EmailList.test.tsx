import { render, screen } from "@testing-library/react";
import { EmailList } from "../../../../components/EmailList";
import { emailItemsList } from "../../../../__mocks__/emailItemsMock";
import "@testing-library/jest-dom";

describe("EmailList.tsx", () => {
  it("Receive multiple Email elements and render them in a table element", () => {
    render(<EmailList emails={emailItemsList} />);

    const table = screen.getByRole("table");
    expect(table.childElementCount).toBe(1);
    expect(table.children[0].childElementCount).toBe(emailItemsList.length);
  });

  it("UI layout and design are remain the same", () => {
    const { container } = render(<EmailList emails={emailItemsList} />);
    expect(container).toMatchSnapshot();
  });
});
