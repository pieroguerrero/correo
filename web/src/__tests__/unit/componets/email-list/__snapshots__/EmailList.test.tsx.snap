// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailList.tsx UI layout and design are remain the same 1`] = `
<div>
  <table
    class="w-full h-full"
  >
    <tbody>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
      <tr
        class="flex flex-row items-center space-x-4 py-2 px-4 max-w-full hover:bg-gray-100 border-b-[1px] border-b-gray-200"
      >
        <td
          class="flex items-center"
        >
          <input
            type="checkbox"
          />
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-4 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.0489 4.92705C11.3483 4.00574 12.6517 4.00574 12.9511 4.92705L14.2451 8.90983C14.379 9.32185 14.763 9.60081 15.1962 9.60081H19.3839C20.3527 9.60081 20.7554 10.8404 19.9717 11.4098L16.5838 13.8713C16.2333 14.126 16.0866 14.5773 16.2205 14.9894L17.5146 18.9721C17.8139 19.8934 16.7595 20.6596 15.9757 20.0902L12.5878 17.6287C12.2373 17.374 11.7627 17.374 11.4122 17.6287L8.02426 20.0902C7.24054 20.6596 6.18607 19.8934 6.48542 18.9721L7.7795 14.9894C7.91338 14.5773 7.76672 14.126 7.41623 13.8713L4.02827 11.4098C3.24456 10.8404 3.64734 9.60081 4.61606 9.60081H8.8038C9.23703 9.60081 9.62099 9.32185 9.75486 8.90983L11.0489 4.92705Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex items-center"
        >
          <button>
            <svg
              class="w-5 h-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m3 19 4.5-7L3 5h12q.5 0 .938.225.437.225.712.625L21 12l-4.35 6.15q-.275.4-.712.625Q15.5 19 15 19Zm3.65-2H15l3.55-5L15 7H6.65l3.25 5Zm3.25-5L6.65 7l3.25 5-3.25 5Z"
              />
            </svg>
          </button>
        </td>
        <td
          class="flex-1 shrink-0 flex items-center"
        >
          <button
            class="flex flex-row items-center space-x-4 w-full"
          >
            <div
              class="w-[200px] text-left"
            >
              <p
                class="text-sm font-bold "
              >
                Bakare Tunde
              </p>
            </div>
            <div
              class="flex flex-1 mr-7"
            >
              <p
                class="text-sm font-bold"
              >
                An importan donation for you
              </p>
              <p
                class=" text-sm font-light text-gray-500"
              >
                 - REQUEST FOR ASSISTANCE-STRICTLY CONFIDENTIAL
              </p>
            </div>
            <div>
              <p
                class="text-sm font-semibold"
              >
                Mar 15
              </p>
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
